sdk/react/widgets/CastPlayPauseButtons.js

```diff
@@ -0,0 +1,352 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {
+  ActivityIndicator, Animated, TouchableHighlight, View,
+} from 'react-native';
+
+import SkipButton from './SkipButton';
+import SwitchButton from './SwitchButton';
+import { BUTTON_NAMES, VALUES } from '../constants';
+
+const timerForSkipButtons = require('react-native-timer');
+const Utils = require('../utils');
+const AccessibilityUtils = require('../accessibilityUtils');
+
+// Uses the rectbutton styles
+const styles = require('../utils')
+  .getStyles(require('./style/RectButtonStyles.json'));
+
+const PLAY = 'play';
+const PAUSE = 'pause';
+const FORWARD = 'seekForward';
+const NEXT = 'next';
+const PREVIOUS = 'previous';
+const BACKWARD = 'seekBackward';
+
+class CastPlayPauseButtons extends React.Component {
+  static propTypes = {
+    seekEnabled: PropTypes.bool.isRequired,
+    ffActive: PropTypes.bool.isRequired,
+    icons: PropTypes.object.isRequired,
+    position: PropTypes.string.isRequired,
+    onPress: PropTypes.func.isRequired,
+    onSeekPressed: PropTypes.func.isRequired,
+    onSwitchPressed: PropTypes.func.isRequired,
+    seekForwardValue: PropTypes.number.isRequired,
+    seekBackwardValue: PropTypes.number.isRequired,
+    opacity: PropTypes.number.isRequired,
+    frameWidth: PropTypes.number.isRequired,
+    frameHeight: PropTypes.number.isRequired,
+    buttonWidth: PropTypes.number.isRequired,
+    buttonHeight: PropTypes.number.isRequired,
+    buttonColor: PropTypes.string.isRequired,
+    buttonStyle: PropTypes.object.isRequired,
+    fontSize: PropTypes.number.isRequired,
+    style: PropTypes.object.isRequired,
+    showButton: PropTypes.bool.isRequired,
+    showSeekButtons: PropTypes.bool.isRequired,
+    playing: PropTypes.bool.isRequired,
+    loading: PropTypes.bool.isRequired,
+    initialPlay: PropTypes.bool.isRequired,
+    config: PropTypes.object.isRequired
+  };
+
+  state = {
+    playPause: {
+      animationScale: new Animated.Value(1),
+      animationOpacity: new Animated.Value(1),
+    },
+    skipButtons: {
+      animationScale: new Animated.Value(1),
+      animationOpacity: new Animated.Value(1),
+    },
+    playing: false,
+    skipCount: 0,
+  };
+
+  componentWillMount() {
+    const { playing } = this.props;
+    this.state.playing = playing;
+    this.state.skipButtons.animationOpacity.setValue(1);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { playing } = this.props;
+
+    if (nextProps.playing !== playing) {
+      this.state.playing = nextProps.playing;
+    }
+  }
+
+  componentWillUnmount() {
+    timerForSkipButtons.clearTimeout(this);
+  }
+
+  onPress() {
+    const { showButton, onPress } = this.props;
+    if (showButton) {
+      onPress(BUTTON_NAMES.PLAY_PAUSE);
+    } else {
+      onPress(BUTTON_NAMES.RESET_AUTOHIDE);
+    }
+  }
+
+  onSkipPress(isForward) {
```

Comment:
could you add jsdoc to describe what's going on here?

Reply:
done. Thanks a lot for you review!